embed/etcd.go

```diff
@@ -82,7 +82,7 @@ type Etcd struct {
 type peerListener struct {
 	net.Listener
 	serve func() error
-	close func(context.Context) error
+	close func(time.Duration) error
```

Comment:
https://golang.org/pkg/net/http/#Server.Shutdown takes context. i am not sure why we want to change to time.duration.